wiki/development/coding-guidelines.md

```diff
@@ -522,229 +487,193 @@ if (isnil "_return") exitwith {};
+|`HASHLIST_CREATELIST(keys)` | Creates a new hashlist with the default keys, pass [] for no default keys. |
+|`HASHLIST_CREATEHASH(hashlist)` | Returns a blank hash template from a hashlist. |
+|`HASHLIST_PUSH(hashList, hash)` | Pushes a new hash onto the end of the list. |
+|`HASHLIST_SELECT(hashlist, index)` | Returns the hash at that index in the list. |
+|`HASHLIST_SET(hashlist, index, hash)` | Sets a specific index to that hash. |
 
 ##### 7.4.1.1. A note on pass by reference and hashes
 
-Hashes and hashlists are implemented with SQF arrays, and as such they are passed by reference to other functions. Remember to make copies (using the + operator) if you intend for the hash or hashlist to be modified with out the need for changing the original value.
+Hashes and hashlists are implemented with SQF arrays, and as such they are passed by reference to other functions. Remember to make copies (using the `+` operator) if you intend for the hash or hashlist to be modified with out the need for changing the original value.
 
 ## 8. Performance Considerations
 
 ### 8.1. Adding Elements to Arrays
-When adding new elements to an array, pushback shall be used instead of the binary addition or set. When adding multiple elements to an array `append` may be used instead of pushback.
+When adding new elements to an array, `pushBack` shall be used instead of the binary addition or `set`. When adding multiple elements to an array `append` may be used instead.
 
 Good:
-
 ```js
-_a pushback _value;
+_a pushBack _value;
 ```
 
 Also good:
-
 ```js
 _a append [1,2,3];
 ```
 
 Bad:
-
 ```js
 _a set [ count _a, _value];
 _a = a + _[value];
 ```
 
-When adding an new element to a dynamic location in an array or when the index is precaluclated, set may be used.
+When adding an new element to a dynamic location in an array or when the index is pre-calculated, `set` may be used.
 
 When adding multiple elements to an array, the binary addition may be used for the entire addition.
 
-### 8.2. CreateVehicle
-CreateVehicle array shall be used instead of createVehicle
+### 8.2. `createVehicle`
+`createVehicle` array shall be used.
 
 ### 8.3. Unscheduled vs Scheduled
-All code that has a visable effect for the user or requires time specific guaranteed excution shall be written in unscheduled space.
+All code that has a visible effect for the user or requires time specific guaranteed execution shall be written in unscheduled space.
 
-### 8.4. Avoid Spawn & execVM
-ExecVM and spawn are to be avoided wherever possible.
+### 8.4. Avoid `spawn` & `execVM`
+`execVM` and `spawn` are to be avoided wherever possible.
 
 ### 8.5. Empty Arrays
-When checking if an array is empty, you can use both count or isEqualTo.
+When checking if an array is empty `isEqualTo` shall be used.
 
-### 8.6. For Loops
+### 8.6. `for` Loops
 
 ```js
 for "_y" from # to # step # do { ... }
 ```
-
 should be used instead of
```

Comment:
I was about to argue against shall, but after reading a bit about it I believe @bux578 is right: shall is preferred when stating rules.